test/sql/pg_dump.sql

```diff
@@ -2,10 +2,42 @@
 -- Please see the included NOTICE for copyright information and
 -- LICENSE-APACHE for a copy of the license.
 
+\set TEST_DBNAME_EXTRA :TEST_DBNAME _extra
+
 \o /dev/null
 \ir include/insert_two_partitions.sql
 \o
 \c :TEST_DBNAME :ROLE_SUPERUSER
+
+CREATE VIEW worker_counts AS
+WITH
+  tsdb_workers AS (SELECT * FROM pg_stat_activity WHERE application_name LIKE 'TimescaleDB%')
+SELECT count(*) FILTER (WHERE application_name LIKE '%Scheduler') AS scheduler_count,
+       count(*) FILTER (WHERE application_name LIKE '%Launcher') AS launcher_count
+  FROM tsdb_workers WHERE datname = :'TEST_DBNAME';
+
+CREATE FUNCTION wait_worker_counts(scheduler INTEGER, launcher INTEGER)
+RETURNS BOOLEAN LANGUAGE PLPGSQL AS
+$BODY$
+DECLARE
+	r INTEGER;
+BEGIN
+    FOR i in 1..10
+    LOOP
+       SELECT COUNT(*) INTO r FROM worker_counts
+       WHERE launcher = launcher_count
+	 AND scheduler = scheduler_count;
+       IF r < 1 THEN
+	  PERFORM pg_sleep(1);
+	  PERFORM pg_stat_clear_snapshot();
```

Comment:
Add comment on why clearing stats snapshot  is needed. It wasn't obvious to me so had to look it up.